Added Maps integration

Lukerob97 opened the pull request:
MAPS INTEGRATION
DONT MERGE UNTIL MULTIPLE PEOPLE HAVE REVIEWED-  LARGE CHANGES

DETAILS
-increased minSDK version for play services compability
-Added button to main activity labelled Find Stores - this can be temporary
-Added MapsActivity, the button on main switches to this activity
-Added google maps support
-Model package is used to parse the .json retrieved using the API key, this information can in future be used to improve the maps to include images, opening hours etc of each store
-Remote package is used for retrieving the .json through the API web request
-currently there are temporary icons being used located in drawable package, these will be replaced with store icons
-bottom_menu is used for the bar at the bottom of the map activity, this can be temporary and removed at a future date
-added API keys to strings.xml in values package
-edited AndroidManifest.xml to include the maps activity aswell as the permission requests to access the devices location and internet services
-edited build.gradle to include google play and location services required for map feature

Lukerob97 committed:
Added Maps integration

DETAILS

-Added button to main activity labelled Find Stores - this can be temporary
-Added MapsActivity, the button on main switches to this activity
-Added google maps support
-Model package is used to parse the .json retrieved using the API key, this information can in future be used to improve the maps to include images, opening hours etc of each store
-Remote package is used for retrieving the .json through the API web request
-currently there are temporary icons eing used located in drawable package, these will be replaced with store icons
-bottom_menu is used for the bar at the bottom of the map activity, this can be temporary and removed at a future date
-added API keys to strings.xml in values package
-edited AndroidManifest.xml to include the maps activity aswell as the permission requests to access the devices location and internet services
-edited build.grade to include google play and location services required for map feature

thomastayler commented on app/src/main/res/menu/bottom_menu.xml:
```diff
@@ -0,0 +1,36 @@
+            android:icon="@drawable/ic_restaurant_black_24dp"
+            app:showAsAction="ifRoom"
+            />
+        <!--
```
Is this commented code needed? (Lines 27-35) Or is it for future work?

Lukerob97 replied:
That menu button is for potential back button depending on feedback of UX usability, will keep there for now incase we wish to re-implement it

ZawLinTun16925410 committed:
search machine is added.
item list is added.
total price calculation is added.
SHOPPING_LISTS table is modified by adding another column named "TOTAL_PRICE"

ZawLinTun16925410 committed:
search machine is added.
item list is added.
total price calculation is added.
SHOPPING_LISTS table is modified by adding another column named "TOTAL_PRICE"

ZawLinTun16925410 committed:
Merge branch 'Staging' into ZawLinTun16925410-patch-1

ZawLinTun16925410 committed:
Merge pull request #8 from bestprice2019/ZawLinTun16925410-patch-1

Zaw lin tun16925410 patch 1

Lukerob97 committed:
Merge branch 'Staging' into lukerobinson-patch1

thomastayler commented:
This is been closed as it has been merged at a later date
